src/components/UserInputBlock.tsx

```diff
@@ -29,27 +29,34 @@ interface InputFormValues {
   response: string
 }
 
-export function buildClassName(formDisabled: boolean): string {
+export function buildClassName(formDisabled: boolean, errorResponse: string | undefined): string {
   let className = 'os-form-control'
   if (formDisabled) {
     className += ' os-textarea-disabled'
   }
+  if (errorResponse !== undefined) {
+    className += ' os-wrong-answer-choice'
+  }
   return className
 }
 
 export const UserInputBlock = ({ content, prompt, ack, waitForEvent, fireEvent, buttonText, contentId, onInputSubmitted }: UserInputBlockProps): JSX.Element => {
   const [responseSubmitted, setResponseSubmitted] = useState(false)
   const contentLoadedContext = useContext(ContentLoadedContext)
+  const NON_EMPTY_VALUE_ERROR = 'Please provide valid input'
 
   const schema = Yup.object({
     response: Yup.string()
-      .trim()
-      .required('Please provide valid input')
       .max(MAX_CHARACTER_INPUT_BLOCK_LENGTH, 'Input is too long')
   })
 
-  const handleSubmit = async (values: InputFormValues): Promise<void> => {
-    setResponseSubmitted(true)
+  const handleSubmit = async (values: InputFormValues, { setFieldError }: FormikHelpers<InputFormValues>): Promise<void> => {
+    if (values.response.trim() === '') {
+      setFieldError('response', NON_EMPTY_VALUE_ERROR)
+    } else {
+      setResponseSubmitted(true)
+    }
+
```

Comment:
This is problematic since we continue on if there is an error and do things like fire events and invoke the `onInputSubmitted` handler. This would be analogous to unintentionally burning an attempt on the input problem component. One way to address this would be to use the `return` early pattern similar to what we're doing in the `InputProblem.tsx`:

```javascript
    if (values.response.trim() === '') {
      setFieldError('response', NON_EMPTY_VALUE_ERROR)
      return
    }

    setResponseSubmitted(true)
```

Reply:
Thanks for catching this! I forgot to bring over the early return from the conditional logic used in `InputProblem.tsx`. I've updated accordingly.